br/pkg/lightning/backend/local/local.go

```diff
@@ -1073,28 +1073,26 @@ func readAndSplitIntoRange(
 	sizeLimit int64,
 	keysLimit int64,
 ) ([]common.Range, error) {
-	firstKey, lastKey, err := engine.GetKeyRange()
+	startKey, endKey, err := engine.GetKeyRange()
 	if err != nil {
 		return nil, err
 	}
-	if firstKey == nil {
+	if startKey == nil {
 		return nil, errors.New("could not find first pair")
 	}
 
-	endKey := nextKey(lastKey)
```

Comment:
seems we have do it previously, what's the difference by move it into `GetKeyRange`?

Reply:
To make `GetKeyRange` has unified logic for local engine and external engine. For external engine, it's more easier to record end key rather than max key (last key) when generating external engine. The wrong usage of end key with max key causes the bug of 11100001 rows